tests/js/specs/letterheads-select.spec.js

```diff
@@ -0,0 +1,197 @@
+
+        $letterheadSelect = $(':contains("Select Letterhead")').parents('tr').find('select');
+        $messageEditor = $('.crm-form-wysiwyg');
+
+        crmWysiwyg._create($messageEditor);
+        crmWysiwyg.setVal(
+          $messageEditor,
+          'Example content'
+        );
+        $letterheadSelect.val('2');
+        $letterheadSelect.trigger('change');
+      });
+
+      it('appends the letterhead to the message editor', () => {
+        expect(crmWysiwyg.getVal($messageEditor))
+          .toContain('<p>Welsh Letterhead</p>');
+      });
+
+      it('does not remove any existing content', () => {
+        expect(crmWysiwyg.getVal($messageEditor))
+          .toContain('Example content');
+      });
+    });
+
+    describe('when selecting no letterhead', () => {
+      beforeEach(() => {
+        $('body').append(getEmailFormFixture());
+        $().triggerBlockedOnReadyListeners();
+
+        $letterheadSelect = $(':contains("Select Letterhead")').parents('tr').find('select');
+        $messageEditor = $('.crm-form-wysiwyg');
+
+        crmWysiwyg._create($messageEditor);
+        crmWysiwyg.setVal(
+          $messageEditor,
+          'Example content'
+        );
+        $letterheadSelect.val('2');
+        $letterheadSelect.trigger('change');
+        $letterheadSelect.val('');
+        $letterheadSelect.trigger('change');
+      });
+
+      it('does not contain a letterhead element', () => {
+        expect(crmWysiwyg.getVal($messageEditor))
+          .not.toContain('<p>Welsh Letterhead</p>');
+      });
+
+      it('does not remove any existing content', () => {
+        expect(crmWysiwyg.getVal($messageEditor))
+          .toContain('Example content');
+      });
+    });
+
+    describe('when changing the current template', () => {
+      beforeEach(() => {
+        jasmine.clock().install();
+        $('body').append(getEmailFormFixture());
+
+        $().triggerBlockedOnReadyListeners();
+
+        const $templateSelect = $('[name="template"]');
+        $letterheadSelect = $(':contains("Select Letterhead")').parents('tr').find('select');
+        $messageEditor = $('.crm-form-wysiwyg');
+
+        crmWysiwyg._create($messageEditor);
+        $letterheadSelect.val('2');
+        $letterheadSelect.trigger('change');
+        crmWysiwyg.setVal(
+          $messageEditor,
+          'Template content'
+        );
+        $templateSelect.trigger('change');
+        CKEDITOR.instances.html_message.fire('change');
+        jasmine.clock().tick();
+      });
+
+      afterEach(() => {
+        jasmine.clock().uninstall();
+      });
+
+      it('preserves the letterhead element', () => {
+        expect(crmWysiwyg.getVal($messageEditor))
+          .toContain('<p>Welsh Letterhead</p>');
+      });
+
+      it('does not remove any existing content', () => {
+        expect(crmWysiwyg.getVal($messageEditor))
+          .toContain('Template content');
+      });
+    });
+
+    function getEmailFormFixture() {
```

Comment:
These functions are not documented, maybe the linter is not configured properly?

Reply:
Thanks for spotting it @deb1990 .

The linter has not been properly configured. I'm using a local one that is less strict. Right now there's no time to properly configure the JS lint, but will try to implement it on the next task.